src/de.rs

```diff
@@ -121,33 +145,52 @@ fn parse_length(value: &[u8]) -> StdResult<usize> {
     .into())
 }
 
+/// This will split off the first key from the value based on the provided key length.
+/// Since from_vec expects that the last key is not length prefixed, we need to remove the length prefix.
+/// This should not be called on the last key within a compound key.
+/// The return value is ordered as (first_key, remainder)
+fn split_off_first_key(key_len: u16, value: &[u8]) -> StdResult<(Vec<u8>, &[u8])> {
+    let mut slice_index: usize = 0;
+    let mut first_key = Vec::new();
+    // First iterate over the sub keys
+    for key_index in 0..key_len {
+        // Key length is always 2 bytes
+        let key_start_index = slice_index + 2;
+        let len_slice = &value[slice_index..key_start_index];
+        // If this is not the last key, we need to add the length prefix
+        if key_index != key_len - 1 {
+            first_key.extend_from_slice(len_slice);
+        }
+        let subkey_len = parse_length(len_slice)?;
+        first_key.extend_from_slice(&value[key_start_index..key_start_index + subkey_len]);
```

Comment:
Nit: These extensions can be avoided by computing the full key length in bytes iteratively, and then doing a single vector assignment / split outside the loop.

Reply:
For sure you can limit the number of extensions, it does clutter up the code more though. The alternative would be using `Vec::split_off` twice (which is two allocations), calling Vec::remove twice (not very efficient), or writing a custom loop to remove the length prefix (cumbersome).

I personally would leave as is but if you have a nice way of doing it feel free!

Reply:
Improved naming and vars, but left the main logic intact.